### ISSUE TITLE
CrossEntropyLoss with ignore_index is not working as expected

### ISSUE THREAD

**Issue by ibeltagy** [pytorch/xla#859]
Here's an example to replicate the issue: 

```
import torch
import torch_xla_py.xla_model as xm
loss = torch.nn.CrossEntropyLoss(ignore_index=-1)
labels = torch.LongTensor([-1, -1, 3])
preds = torch.ones(3, 5)
print(loss(preds, labels)) # tensor(1.6094)

xla_device = xm.xla_device()
labels = labels.to(xla_device)
preds = preds.to(xla_device)
print(loss(preds, labels))  # expected: tensor(1.6094, device='xla:1')
                                           # printed: tensor(0.5365, device='xla:1')
```

The loss of each individual item is computed correctly (taking the `ignore_index` flag into account), but then the averaging ignores it

**Comment by ibeltagy** [comment]
Checking the code here: https://github.com/pytorch/xla/blob/c9b69ddcd3414e2a2ce0810d8b0c15056f65bc21/torch_xla/csrc/aten_xla_type.cpp#L2056
If I understand this condition correctly, it is saying 
1) computing the loss with`ignore_index` enabled will run on the CPU (no TPU implementation yet)
2) that `ignore_index` is enabled when `ignore_index >=0`

Did I get that right? If so, 
1) would you be able to add TPU support for `ignore_index`? 
2) the condition `ignore_index >=0` is not correct. `ignore_index` should work for any integer. In my application, I am using `-1` as the label that should be ignored.

**Comment by asuhan** [comment]
It should be quite easy to add TPU support for `ignore_index`. I'll have to check the semantics, it's been a while since I implemented that operator, but sounds like it's a bug indeed.

**Comment by dlibenzi** [comment]
If we are not going to fix this soon, let's at least use the pytorch's default -100 as CPU-switch value.

**Comment by ibeltagy** [comment]
Thanks @asuhan. It would be great to add TPU support for `ignore_index`. I am trying to run BERT pretraining on TPUs, and `ignore_index` is needed for the masked language model loss.

**Comment by asuhan** [comment]
@dlibenzi I think I can find some time this week to fix the bug.
